diff --git a/src/widgets/FollowButton.vala b/src/widgets/FollowButton.vala
index 4207e7f..48d8df3 100644
--- a/src/widgets/FollowButton.vala
+++ b/src/widgets/FollowButton.vala
@@ -45,7 +45,7 @@ class FollowButton : Gtk.Button {
 
     stack.add (follow_label);
     stack.add (unfollow_label);
-    stack.interpolate_size = true;
+    //stack.interpolate_size = true;
     stack.transition_type = Gtk.StackTransitionType.CROSSFADE;
     stack.hhomogeneous = false;
     stack.vhomogeneous = true;
