From 2892057c0797eb2cdfa751cdc2e0ccf96a4c5025 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Claudio=20Andr=C3=A9?= <claudioandre.br@gmail.com>
Date: Thu, 20 Apr 2017 22:36:12 -0300
Subject: [PATCH 2/2] maintenance: fix the expected data type size

---
 src/memdbg.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/memdbg.c b/src/memdbg.c
index f7551f5e1..2cadca801 100644
--- a/src/memdbg.c
+++ b/src/memdbg.c
@@ -545,7 +545,8 @@ void * MEMDBG_calloc(size_t count, size_t size, char *file, int line)
  */
 void * MEMDBG_alloc(size_t size, char *file, int line)
 {
-	MEMDBG_HDR      *p, *p2;
+	MEMDBG_HDR      *p;
+	void *p2;
 
 	if ( ((signed long long)mem_size) < 0)
 		fprintf(stderr, "MEMDBG_alloc "LLd" %s:%d  mem:"LLd"\n", (unsigned long long)size, file, line, (unsigned long long)mem_size);
@@ -614,7 +615,8 @@ void * MEMDBG_alloc(size_t size, char *file, int line)
  */
 void * MEMDBG_alloc_align(size_t size, int align, char *file, int line)
 {
-	MEMDBG_HDR      *p, *p2;
+	MEMDBG_HDR      *p;
+	void *p2;
 	char *p3;
 
 	if ( ((signed long long)mem_size) < 0)
-- 
2.11.0

